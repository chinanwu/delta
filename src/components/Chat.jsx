import React from 'react';
import PropTypes from 'prop-types';

import './Chat.less';

export const Chat = ({ messages, text, onChange, onKeyDown, onClick }) => (
	<div className="Chat">
		<div className="Chat__messages">
			{messages.map((message, index) => (
				<div className="Chat__message" key={index}>
					{message}
				</div>
			))}
		</div>
		<div className="Chat__inputBar">
			<input
				id="chatInput"
				className="Chat__input"
				type="text"
				value={text}
				onChange={onChange}
				onKeyDown={onKeyDown}
			/>
			<button id="chatInputBtn" className="Chat__inputBtn" onClick={onClick}>
				Submit
			</button>
		</div>
	</div>
);

Chat.propTypes = {
	messages: PropTypes.array,
	text: PropTypes.string,
	onChange: PropTypes.func,
	onKeyDown: PropTypes.func,
	onClick: PropTypes.func,
};

export default Chat;
