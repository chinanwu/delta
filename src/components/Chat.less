@import '../style/styles';

.Chat {
	border: 1px solid black;
	height: 100%;
	position: relative;
	margin-left: 18px;
	margin-right: 18px;
	margin-bottom: 18px;
}

.Chat__messages {
	display: flex;
	flex-direction: column;
	overflow: scroll;
	height: 90%;
	box-sizing: border-box;
}

.Chat__message {
	margin-left: 6px;
}

.Game__history:first-child {
	margin-top: 6px;
}

.Chat__inputBar {
	border-top: 1px solid black;
	display: flex;
	justify-content: center;
	align-items: center;
	height: 10%;
}

.Chat__input {
	margin-right: 6px;
	margin-left: 12px;
	font-size: 15px;
}

.Chat__inputBtn {
	@greenBtn();
	padding: 6px;
	margin-right: 12px;
}

.Chat__inputBtn:hover {
	@greenBtnHover();
}

.Chat__inputBtn:active {
	@greenBtnActive();
}
