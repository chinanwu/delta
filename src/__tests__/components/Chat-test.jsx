import { shallow } from 'enzyme';
import React from 'react';

import { Chat } from '../../components/Chat.jsx';

jest.unmock('../../components/Chat.jsx');

describe('Chat component', () => {
	describe('rendering', () => {
		it('renders', () => {
			const wrapper = shallow(<Chat messages={['test']} />);
			expect(wrapper.getElement()).toMatchSnapshot();
		});
	});

	describe('behaviour', () => {
		it('calls onChange when input changes', () => {
			const onChange = jest.fn();
			const wrapper = shallow(<Chat messages={[]} onChange={onChange} />);
			wrapper.find('#chatInput').simulate('change');
			expect(onChange).toHaveBeenCalled();
		});

		it('calls onKeyDown when key down', () => {
			const onKeyDown = jest.fn();
			const wrapper = shallow(<Chat messages={[]} onKeyDown={onKeyDown} />);
			wrapper.find('#chatInput').simulate('keydown');
			expect(onKeyDown).toHaveBeenCalled();
		});

		it('calls onClick when click on button', () => {
			const onClick = jest.fn();
			const wrapper = shallow(<Chat messages={[]} onClick={onClick} />);
			wrapper.find('#chatInputBtn').simulate('click');
			expect(onClick).toHaveBeenCalled();
		});
	});
});
