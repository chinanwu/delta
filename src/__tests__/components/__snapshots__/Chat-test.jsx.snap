// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chat component rendering renders 1`] = `
<div
  className="Chat"
>
  <div
    className="Chat__messages"
  >
    <div
      className="Chat__message"
    >
      test
    </div>
  </div>
  <div
    className="Chat__inputBar"
  >
    <input
      className="Chat__input"
      id="chatInput"
      type="text"
    />
    <button
      className="Chat__inputBtn"
      id="chatInputBtn"
    >
      Submit
    </button>
  </div>
</div>
`;
